src/Dapr.PluggableComponents.Tests/Adaptors/PubSubAdaptorTests.cs

```diff
@@ -0,0 +1,379 @@
+    {
+        var mockStateStore = new Mock<IPubSub>(MockBehavior.Strict);
+
+        var adaptor = AdaptorFixture.CreatePubSub(mockStateStore.Object);
+
+        using var context = new TestServerCallContext();
+
+        await adaptor.Ping(
+            new PingRequest(),
+            context);
+    }
+
+    [Fact]
+    public async Task PingWithLiveness()
+    {
+        var mockStateStore = new Mock<IPubSub>();
+        var mockLiveness = mockStateStore.As<IPluggableComponentLiveness>();
+
+        mockLiveness
+            .Setup(component => component.PingAsync(It.IsAny<CancellationToken>()));
+
+        var adaptor = AdaptorFixture.CreatePubSub(mockStateStore.Object);
+
+        using var context = new TestServerCallContext();
+
+        await adaptor.Ping(
+            new PingRequest(),
+            context);
+
+        mockLiveness
+            .Verify(
+                component => component.PingAsync(
+                    It.Is<CancellationToken>(token => token == context.CancellationToken)),
+                Times.Once());
+    }
+
+    [Fact]
+    public async Task FeaturesWithNoFeatures()
+    {
+        var mockStateStore = new Mock<IPubSub>(MockBehavior.Strict);
+
+        var adaptor = AdaptorFixture.CreatePubSub(mockStateStore.Object);
+
+        using var context = new TestServerCallContext();
+
+        var response = await adaptor.Features(
+            new FeaturesRequest(),
+            context);
+
+        Assert.NotNull(response);
+        Assert.Empty(response.Features);
+    }
+
+    [Fact]
+    public async Task FeaturesWithFeatures()
+    {
+        var mockStateStore = new Mock<IPubSub>();
+        var mockFeatures = mockStateStore.As<IPluggableComponentFeatures>();
+
+        mockFeatures
+            .Setup(component => component.GetFeaturesAsync(It.IsAny<CancellationToken>()))
+            .ReturnsAsync(new[] { "feature1", "feature2" });
+
+        var adaptor = AdaptorFixture.CreatePubSub(mockStateStore.Object);
+
+        using var context = new TestServerCallContext();
+
+        var response = await adaptor.Features(
+            new FeaturesRequest(),
+            context);
+
+        Assert.NotNull(response);
+        Assert.Equal(2, response.Features.Count);
+        Assert.Contains("feature1", response.Features);
+        Assert.Contains("feature2", response.Features);
+
+        mockFeatures
+            .Verify(
+                component => component.GetFeaturesAsync(
+                    It.Is<CancellationToken>(token => token == context.CancellationToken)),
+                Times.Once());
+    }
+
+    [Fact]
+    public async Task Publish()
+    {
+        var mockStateStore = new Mock<IPubSub>();
+
+        mockStateStore
+            .Setup(component => component.PublishAsync(It.IsAny<PubSubPublishRequest>(), It.IsAny<CancellationToken>()));
+
+        var adaptor = AdaptorFixture.CreatePubSub(mockStateStore.Object);
+
+        using var context = new TestServerCallContext();
```

Comment:
Just wondering if, aside the mock configuration, its definition, the adaptor creation and the context could be part of an `[Before]` and `[After]` method or equivalent. 

Aside the only test where the mock is created as strict, every other one does the same steps.

Reply:
That's a good idea.  I had generalized creation of the adaptor itself across tests, but could do the same for context and the mock itself.  I'll do that for the pub-sub tests in this PR, and come back for the state-store tests in a follow up PR.